assembly/src/parser/grammar.lalrpop

```diff
@@ -684,21 +684,32 @@ SystemEvent: Instruction = {
 
 #[inline]
 InstWithErrorCode: Instruction = {
-    "assert" <error_code:MaybeAssertCode> => error_code.map(Instruction::AssertWithError).unwrap_or(Instruction::Assert),
-    "assertz" <error_code:MaybeAssertCode> => error_code.map(Instruction::AssertzWithError).unwrap_or(Instruction::Assertz),
-    "assert_eq" <error_code:MaybeAssertCode> => error_code.map(Instruction::AssertEqWithError).unwrap_or(Instruction::AssertEq),
-    "assert_eqw" <error_code:MaybeAssertCode> => error_code.map(Instruction::AssertEqwWithError).unwrap_or(Instruction::AssertEqw),
-    "u32assert" <error_code:MaybeAssertCode> => error_code.map(Instruction::U32AssertWithError).unwrap_or(Instruction::U32Assert),
-    "u32assert2" <error_code:MaybeAssertCode> => error_code.map(Instruction::U32Assert2WithError).unwrap_or(Instruction::U32Assert2),
-    "u32assertw" <error_code:MaybeAssertCode> => error_code.map(Instruction::U32AssertWWithError).unwrap_or(Instruction::U32AssertW),
-    "mtree_verify" <error_code:MaybeAssertCode> => error_code.map(Instruction::MTreeVerifyWithError).unwrap_or(Instruction::MTreeVerify),
-}
-
-MaybeAssertCode: Option<Immediate<u32>> = {
-    "." "err" "=" <code:ImmValue<U32>> => Some(code),
+    "assert" <error_code:MaybeAssertMessage> => error_code.map(Instruction::AssertWithError).unwrap_or(Instruction::Assert),
+    "assertz" <error_code:MaybeAssertMessage> => error_code.map(Instruction::AssertzWithError).unwrap_or(Instruction::Assertz),
+    "assert_eq" <error_code:MaybeAssertMessage> => error_code.map(Instruction::AssertEqWithError).unwrap_or(Instruction::AssertEq),
+    "assert_eqw" <error_code:MaybeAssertMessage> => error_code.map(Instruction::AssertEqwWithError).unwrap_or(Instruction::AssertEqw),
+    "u32assert" <error_code:MaybeAssertMessage> => error_code.map(Instruction::U32AssertWithError).unwrap_or(Instruction::U32Assert),
+    "u32assert2" <error_code:MaybeAssertMessage> => error_code.map(Instruction::U32Assert2WithError).unwrap_or(Instruction::U32Assert2),
+    "u32assertw" <error_code:MaybeAssertMessage> => error_code.map(Instruction::U32AssertWWithError).unwrap_or(Instruction::U32AssertW),
+    "mtree_verify" <error_code:MaybeAssertMessage> => error_code.map(Instruction::MTreeVerifyWithError).unwrap_or(Instruction::MTreeVerify),
+}
```

Comment:
This change still preserves the bug described in #1764. Could we try to fix it while we're making a change here anyways? Assuming this is not too much work of course.

cc @bitwalker

Reply:
I'll address that in a separate PR @plafer, as that needs to happen in the lexer, not the parser (and we aren't changing the lexer here).